pkg/cache/context_recovery.go

```diff
@@ -80,86 +73,77 @@ func (ctx *Context) recover(mgr []interfaces.Recoverable, due time.Duration) err
 		ctx.nodes.addAndReportNode(node, false)
 	}
 
-	// current, disable getting pods for a node during test,
-	// because in the tests, we don't really send existing allocations
-	// we simply simulate to accept or reject nodes on conditions.
-	if !ctx.apiProvider.IsTestingMode() {
-		var podList *corev1.PodList
-		podList, err = ctx.apiProvider.GetAPIs().KubeClient.GetClientSet().
-			CoreV1().Pods("").
-			List(context.Background(), metav1.ListOptions{})
-		if err != nil {
-			return err
-		}
+	pods, err := ctx.apiProvider.GetAPIs().PodInformer.Lister().List(labels.Everything())
```

Comment:
I rewrote this to `PodInformer.Lister().List()`, documentation says it's faster.
Although at this point, we already have the list of pods from a previous listing, so we can even skip this (perhaps in a separate JIRA?).